tests/functional/DatadirTest.php

```diff
@@ -67,5 +74,63 @@ protected function setUp(): void
             ],
         ];
         JsonHelper::writeFile(__DIR__ . '/manual-recursion/expected/data/out/state.json', $state);
+
+        // --- only-new-files tests ---
+        $inputState = [
+            "ex-ftp-state" => [
+                "newest-timestamp" => 0,
+                "last-timestamp-files" => [],
+            ],
+        ];
+        $outputState = [
+            "ex-ftp-state" => [
+                "newest-timestamp" => $timestamps["file_1.txt"],
+                "last-timestamp-files" => ["file_1.txt", "Zvlášť zákeřný učeň s ďolíčky běží podél zóny úlů.csv"],
+            ],
+        ];
+        JsonHelper::writeFile(__DIR__ . '/only-new-files/expected/data/out/state.json', $outputState);
+        JsonHelper::writeFile(__DIR__ . '/only-new-files/source/data/in/state.json', $inputState);
+
+        // -- new-files-from-old-state test --
+        $inputState = [
+            "ex-ftp-state" => [
+                "newest-timestamp" => $timestamps["file_1.txt"],
+                "last-timestamp-files" => ["file_1.txt", "Zvlášť zákeřný učeň s ďolíčky běží podél zóny úlů.csv"],
+            ],
+        ];
+        JsonHelper::writeFile(__DIR__ . '/new-files-from-old-state/source/data/in/state.json', $inputState);
+    }
+
+    /**
+     * @dataProvider provideDatadirSpecifications
+     */
+    public function testDatadir(DatadirTestSpecificationInterface $specification): void
+    {
+        $tempDatadir = $this->getTempDatadir($specification);
+
+        $sourceDatadir = $specification->getSourceDatadirDirectory();
+
+        if (in_array('new-files-from-old-state', explode('/', $sourceDatadir))) {
```

Comment:
Could it be encapsulated in its own method?

Reply:
![Snímek obrazovky 2019-10-15 v 13 12 15](https://user-images.githubusercontent.com/1696274/66849569-7a024480-ef4d-11e9-898a-53592e02a43c.png)

Reply:
fixed, diky